### PR TITLE
[PXN-2943] - Implementation of unit tests

### PR DESCRIPTION
**Pull request by samantaalbanez**
**Changes:**

- Adjustment of existing tests;
- Converting classes from java to kotlin;
- Implementation of new tests;
- Changes and implementations were performed using mockk;
- Adjustment in class TestBase;

**Changed packages:**
- _carddrawer/model_

### CODE REVIEW

File: carddrawer/src/test/java/com/meli/android/carddrawer/TestBase.kt [hunk 1]
```diff
@@ -1,11 +1,11 @@
-package com.meli.android.carddrawer.configuration.base
+package com.meli.android.carddrawer
 
 import android.content.Context
 import io.mockk.MockKAnnotations
 import io.mockk.impl.annotations.MockK
 import org.junit.Before
 
-open class ConfigurationTestBase {
+open class TestBase {
```
Comment (guchito9) [inline carddrawer/src/test/java/com/meli/android/carddrawer/TestBase.kt:8]: I think BaseTest sounds better :P

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardDrawerSourceTest.kt [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import android.widget.ImageView
+import io.mockk.MockKAnnotations
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+
+class CardDrawerSourceTest {
```
Comment (guchito9) [inline carddrawer/src/test/java/com/meli/android/carddrawer/model/CardDrawerSourceTest.kt:12]: This one should extend BaseTest also

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardUITest.kt [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import com.meli.android.carddrawer.configuration.CardDrawerStyle
+import io.mockk.MockKAnnotations
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import io.mockk.verify
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+
+class CardUITest {
```
Comment (guchito9) [inline carddrawer/src/test/java/com/meli/android/carddrawer/model/CardUITest.kt:14]: This one should extend BaseTest also

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardDrawerSourceTest.kt (hunk 2 above)
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/model/CardDrawerSourceTest.kt]: Adjustment made

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardUITest.kt (hunk 3 above)
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/model/CardUITest.kt]: Adjustment made

File: carddrawer/src/test/java/com/meli/android/carddrawer/TestBase.kt (hunk 1 above)
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/TestBase.kt]: Class name changed

File: carddrawer/src/test/java/com/meli/android/carddrawer/configuration/DefaultCardTest.kt [hunk 4]
```diff
@@ -2,14 +2,14 @@ package com.meli.android.carddrawer.configuration
...
-import com.meli.android.carddrawer.configuration.base.ConfigurationTestBase
+import com.meli.android.carddrawer.BaseTest
 import com.meli.android.carddrawer.model.CardAnimationType
 import io.mockk.every
 import org.junit.Assert.assertEquals
 import org.junit.Before
 import org.junit.Test
 
-class DefaultCardConfigurationTest: ConfigurationTestBase() {
+class DefaultCardTest: BaseTest() {
```
Comment (cardosodiego) [inline carddrawer/src/test/java/com/meli/android/carddrawer/configuration/DefaultCardTest.kt:12]: Why we changed the name of class here too?

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+    @Test
+    fun `when getting context of CardAnimator then return value not null`() {
+        initialize()
+        val context = cardAnimator.getDeclaredField("context")
+        Assert.assertNotNull(context)
+    }
+
+    @Test
+    fun `when call function switchView with POSITION_FRON per parameter then fill showingLayout with the same value`() {
```
Comment (cardosodiego) [inline carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt:55]: I think here is POSITION_FRONT in test description, right?

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        val bundle = mockk<Bundle>(relaxed = true)
+        cardAnimator.saveState(bundle)
+        val showingColor: Int = cardAnimator.getDeclaredField("showingColor") as Int
+        verify(exactly = 1) {
+            bundle.putInt("showing_color", showingColor)
+        }
+    }
+
+    @Test
+    fun `when call function showFont then call flipCardFront`() {
```
Comment (cardosodiego) [inline carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt:126]: In test description, i think that is showFront instead showFont, right?

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardConfigurationTest.kt [hunk 7]
```diff
@@ -58,7 +58,39 @@ class CardConfigurationTest {
...
+        val constraintLayout = mockk<ConstraintLayout>()
+        every { cardConfiguration.setUpConstraintLayoutConfiguration(constraintLayout) } answers { callOriginal() }
+        cardConfiguration.setUpConstraintLayoutConfiguration(constraintLayout)
+        verify(exactly = 1) {
+            cardConfiguration.setUpConstraintLayoutConfiguration(constraintLayout)
+        }
+    }
+
+    @Test
+    fun `when call function v then if it call one time`() {
```
Comment (cardosodiego) [inline carddrawer/src/test/java/com/meli/android/carddrawer/model/CardConfigurationTest.kt:87]: What is function "v" ?

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardUITest.kt [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+        val imageView = mockk<ImageView>()
+        every { cardUI.setCardLogoImage(imageView) } answers { callOriginal() }
+        cardUI.setCardLogoImage(imageView)
+        verify (exactly = 1) {
+            cardUI.setCardLogoImage(imageView)
+        }
+    }
+
+    @Test
+    fun `when set ets bank image to the Imageview then call function setBankImage one time`() {
```
Comment (cardosodiego) [inline carddrawer/src/test/java/com/meli/android/carddrawer/model/CardUITest.kt:73]: What is set ets ?

File: carddrawer/src/test/java/com/meli/android/carddrawer/configuration/CardFontFactoryTest.kt [hunk 9]
```diff
@@ -1,9 +1,9 @@
...
 
-import com.meli.android.carddrawer.configuration.base.ConfigurationTestBase
+import com.meli.android.carddrawer.BaseTest
 import org.junit.Test
 
-class CardFontConfigurationFactoryTest : ConfigurationTestBase() {
+class CardFontFactoryTest : BaseTest() {
 
     @Test
     fun `when call function getConfiguration of CardFontConfigurationFactory with FontType DARK_TYPE per parameter then return DarkFontConfiguration`() {
```
[the remark this answers is not in the document]
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/configuration/CardFontFactoryTest.kt]: adjusted

File: carddrawer/src/test/java/com/meli/android/carddrawer/configuration/DarkFontTest.kt [hunk 10]
```diff
@@ -3,14 +3,14 @@ package com.meli.android.carddrawer.configuration
...
-import com.meli.android.carddrawer.configuration.base.ConfigurationTestBase
+import com.meli.android.carddrawer.BaseTest
 import com.meli.android.carddrawer.configuration.shadow.ShadowConfiguration
 import com.meli.android.carddrawer.configuration.shadow.ShadowFontConfiguration
 import io.mockk.mockk
 import io.mockk.verify
 import org.junit.Test
 
-class DarkFontConfigurationTest : ConfigurationTestBase() {
+class DarkFontTest : BaseTest() {
```
[the remark this answers is not in the document]
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/configuration/DarkFontTest.kt]: adjusted

File: carddrawer/src/test/java/com/meli/android/carddrawer/configuration/DefaultCardTest.kt (hunk 4 above)
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/configuration/DefaultCardTest.kt]: adjusted

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt (hunk 5 above)
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt]: adjusted

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt (hunk 6 above)
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt]: adjusted

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardConfigurationTest.kt (hunk 7 above)
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/model/CardConfigurationTest.kt]: adjusted

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt [hunk 11]
```diff
@@ -0,0 +1,175 @@
...
+    fun `when call function switchView with POSITION_BACK per parameter then fill showingLayout with the same value`() {
+        val mock = mockk<CardAnimator>(relaxed = true)
+        every { mock.switchView(FieldPosition.POSITION_BACK) } answers { callOriginal() }
+        mock.switchView(FieldPosition.POSITION_BACK)
+        val showingLayout = mock.getDeclaredField("showingLayout")
+        Assert.assertEquals(showingLayout, FieldPosition.POSITION_BACK)
+    }
+
+    @Test
+    fun `when call function switchViewWithoutAnimation with POSITION_FRONT per parameter fill showingLayout with POSITION_FRONT`() {
```
Comment (cardosodiego) [inline carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt:80]: in this sentence, is missing "then" sentence.

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt [hunk 12]
```diff
@@ -0,0 +1,175 @@
...
+        initialize()
+        cardAnimator.switchViewWithoutAnimation(FieldPosition.POSITION_FRONT)
+        val showingLayout = cardAnimator.getDeclaredField("showingLayout")
+        Assert.assertEquals(showingLayout, FieldPosition.POSITION_FRONT)
+    }
+
+    @Test
+    fun `when call function switchViewWithoutAnimation with POSITION_BACK per parameter fill showingLayout with POSITION_FRONT`() {
```
Comment (cardosodiego) [inline carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt:88]: in this sentence, is missing "then" sentence.

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+        initialize()
+        cardAnimator.switchViewWithoutAnimation(FieldPosition.POSITION_BACK)
+        val showingLayout = cardAnimator.getDeclaredField("showingLayout")
+        Assert.assertEquals(showingLayout, FieldPosition.POSITION_BACK)
+    }
+
+    @Test
+    fun `when call function switchViewWithoutAnimation with 3 per parameter fill showingLayout with 3`() {
```
Comment (cardosodiego) [inline carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt:96]: in this sentence, is missing "then" sentence.

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardUITest.kt (hunk 8 above)
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/model/CardUITest.kt]: adjusted

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt (hunk 11 above)
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt]: adjusted

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt (hunk 12 above)
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt]: adjusted

File: carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt (hunk 13 above)
Comment (samantaalbanez) [reply carddrawer/src/test/java/com/meli/android/carddrawer/model/CardAnimatorTest.kt]: adjusted

**Review by cardosodiego — APPROVED** [review]

**Pull request merged**